### PR TITLE
06-All-Comments-of-an-article

### PR DESCRIPTION
**Pull request by moskgit**
I've used grid to demonstrate. 
Formatted the date display. 
Not using comment ID (It's for internal purposes).

### CODE REVIEW

**Review by robert112233 — CHANGES_REQUESTED** [review]
Great work Mohsin! The comments are nicely separated and you've fetched them well. I think it makes more sense if the comments component is rendered below the article itself, many comments will reference the article and it would be frustrating to go back and forth between them. You should also handle articles with no comments, for example article 37. Perhaps render a message saying "be the first to comment...." etc. 

Please make these changes and submit another PR for review! Thanks :)

File: src/App.jsx [hunk 1]
```diff
@@ -22,6 +23,7 @@ function App() {
       <Route path="/help" element={ <Help /> } />
       <Route path="/settings" element={ <Settings />} />
       <Route path="/articles/:article_id" element={ <SingleArticle /> } />
+      <Route path="/articles/:article_id/comments" element={ <CommentsForAnArticle_List /> } />
```
Comment (popatre) [inline src/App.jsx:26]: Its up to you, but i'd suggest not having a new page for the comments, and just using your <CommentsForAnArticle_List /> component for the `/parricles/:article_id` page.

its slightly confusing at the moment to navigate to a different page that looks the same, but has the comments at the bottom. 

Maybe one to come back to later on though

File: src/Components/Main/Articles/ArticleCard.jsx [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
+            <Link to={`/articles/${article.article_id}`} className="card-title" href="#">
+                <h1 title="CLICK the HEADING to read an article.">{article.title}</h1>
+            </Link>
+            <h2>Topic: {article.topic}</h2>
+            <h3>Author: {article.author}</h3>
+            <p>
+                <img src={`${article.article_img_url}`} alt="Article Cover" className="cardImage" title="CLICK the HEADING to read an article." />
+            </p>
+            <p>{article.body}</p>
+            <p>Article ID: {`${article.article_id}`}</p>
```
Comment (popatre) [inline src/Components/Main/Articles/ArticleCard.jsx:22]: Not sure someone on the website needs to know the article id - you could keep that private and just for you to use as a key - doesnt really need to be displayed

**Review by popatre — APPROVED** [review]
- Excellent work displaying the comments
- Nice work on handling the articles that have no comments
- Lovely stuff creating a separate component for the comments; stops the components getting too big and bloated

I've approved, but i'd have a look at my comment regarding the com,ments page - maybe one to come back to later though

File: src/App.jsx (hunk 1 above)
Comment (moskgit) [reply src/App.jsx:26]: I'll surely look at it, like you said, later. Thank you for pointing out.

File: src/Components/Main/Articles/ArticleCard.jsx (hunk 2 above)
Comment (moskgit) [reply src/Components/Main/Articles/ArticleCard.jsx:22]: I've now removed it. Thanks.

**Pull request merged**